src/pipx/commands/inject.py

```diff
@@ -98,10 +105,19 @@ def inject(
     suffix: bool = False,
 ) -> ExitCode:
     """Returns pipx exit code."""
+    # Combined list of packages
+    packages = list(package_specs)
+    for filename in requirement_files:
+        packages.extend(parse_requirements(filename))
```

Comment:
How about this to de-duplicate?

```suggestion
    # Combined set of packages
    packages = set(package_specs)
    for filename in requirement_files:
        packages = packages | set(parse_requirements(filename))
```

Reply:
That will only de-duplicate by exactly matching the dependency spec strings, including version numbers and white space. I'm not sure there's much value in that. I think that if we're going to de-duplicate, then you'd want to group by the package name and consolidate the version constraints into a single line, which will be very complicated.

Reply:
I've implemented this. It at least removes duplicated package specifications.